src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -409,10 +415,25 @@ export const DatasetBuilderContents = ({ onStateChange }: { onStateChange: OnSta
       }),
       h(ConceptSetSelector, {
         selectedConceptSets,
-        onChange: (conceptSets) => setSelectedConceptSets(conceptSets),
+        onChange: async (conceptSets) => {
+          setSelectedConceptSets(conceptSets);
+          const uniqueDomains = _.flow(
+            _.flatMap((headerAndValues: HeaderAndValues<ConceptSet>) => headerAndValues.values),
+            _.map((conceptSet: ConceptSet) => conceptSet.domain),
+            _.uniq,
+            _.sortBy((domain) => domain)
+          )(conceptSets);
+          const domainValuesList = await DatasetBuilder().getValuesFromDomains(datasetId, uniqueDomains);
```

Comment:
This refetches from every domain every time. That might not be what we want to do, but I think we're unsure that this is an ajax call long run anyway so I didn't want to worry too much about ironing it all out.

Reply:
I think that the values (columns) are fixed for all feature types and can be fetched once. Even if a user defines their own concept set, it's based on one of the predefined domains (like Condition, Procedure, Drug, etc) and the columns are the same for any subtype of those types.